src/charm.py

```diff
@@ -113,85 +106,125 @@ def _on_jenkins_pebble_ready(self, event: PebbleReadyEvent) -> None:
         container.replan()
         try:
             jenkins.wait_ready()
-            self.unit.status = MaintenanceStatus("Configuring Jenkins.")
+            self.unit.status = ops.MaintenanceStatus("Configuring Jenkins.")
             jenkins.bootstrap(container, self.state.proxy_config)
             # Second Jenkins server start restarts Jenkins to bypass Wizard setup.
             container.restart(self.state.jenkins_service_name)
             jenkins.wait_ready()
         except TimeoutError as exc:
             logger.error("Timed out waiting for Jenkins, %s", exc)
-            self.unit.status = BlockedStatus("Timed out waiting for Jenkins.")
+            self.unit.status = ops.BlockedStatus("Timed out waiting for Jenkins.")
             return
         except jenkins.JenkinsBootstrapError as exc:
             logger.error("Error installing plugins, %s", exc)
-            self.unit.status = BlockedStatus("Error installling plugins.")
+            self.unit.status = ops.BlockedStatus("Error installling plugins.")
             return
 
         try:
             version = jenkins.get_version()
         except jenkins.JenkinsError as exc:
             logger.error("Failed to get Jenkins version, %s", exc)
-            self.unit.status = BlockedStatus("Failed to get Jenkins version.")
+            self.unit.status = ops.BlockedStatus("Failed to get Jenkins version.")
             return
 
         self.unit.set_workload_version(version)
-        self.unit.status = ActiveStatus()
+        self.unit.status = ops.ActiveStatus()
 
-    def _on_get_admin_password(self, event: ActionEvent) -> None:
+    def _on_get_admin_password(self, event: ops.ActionEvent) -> None:
         """Handle get-admin-password event.
 
         Args:
             event: The event fired from get-admin-password action.
         """
-        if not self._jenkins_container.can_connect():
+        container = self.unit.get_container(self.state.jenkins_service_name)
+        if not container.can_connect():
             event.defer()
             return
-        credentials = jenkins.get_admin_credentials(self._jenkins_container)
+        credentials = jenkins.get_admin_credentials(container)
         event.set_results({"password": credentials.password})
 
-    def _on_update_status(self, _: UpdateStatusEvent) -> None:
-        """Handle update status event.
+    def _remove_unlisted_plugins(self, container: ops.Container) -> ops.StatusBase:
+        """Remove plugins that are installed but not allowed.
```

Comment:
s/plugins that are installed but not allowed/installed but not allowed plugins
The simpler the grammar, the easier to read :)

Reply:
Definately, thank you for the suggestion!